railties/lib/rails/generators/actions.rb

```diff
@@ -284,14 +285,15 @@ def log(*args) # :doc:
         # based on the executor parameter provided.
         def execute_command(executor, command, options = {}) # :doc:
           log executor, command
-          env  = options[:env] || ENV["RAILS_ENV"] || "development"
+          env = options[:env] || ENV["RAILS_ENV"] || "development"
+          rails_env = " RAILS_ENV=#{env}" unless options[:without_rails_env]
           sudo = options[:sudo] && !Gem.win_platform? ? "sudo " : ""
           config = { verbose: false }
 
           config[:capture] = options[:capture] if options[:capture]
           config[:abort_on_failure] = options[:abort_on_failure] if options[:abort_on_failure]
 
-          in_root { run("#{sudo}#{extify(executor)} #{command} RAILS_ENV=#{env}", config) }
+          in_root { run("#{sudo}#{extify(executor)} #{command}#{rails_env}", config) }
```

Comment:
Could we fix this by prefixing the command with the Rails environment? Seems off that the environment handling leaks into the generate method.

Reply:
Wait, so if I run a rake task and set an env variable at the end it’s actually not bash that makes it available in ENV, but Rake? I did not know that!

Reply:
Yeah, the shell assumes nothing about program's arguments, no matter whether they include an equal sign :smiley: 

```shell
$ touch a.txt; cat a.txt RAILS_ENV=production
cat: 'RAILS_ENV=production': No existe el archivo o el directorio
```

Reply:
@deivid-rodriguez 
Thanks for your useful information. I was wondering if this feature works on Windows. By doing this, I think that rake was working as expected on Windows.  

@kaspth 
In my understanding, the same fix as rake is necessary for rails command. On unix, there is no problem if you specify the environment variable before the command, but on Windows it is aware that rake and rails are necessary to provide equivalent functionality. What do you think?

Reply:
@y-yagi That's a very good point :+1:.

Another idea would be to add support for passing env to [thor's run method](https://github.com/erikhuda/thor/blob/master/lib/thor/actions.rb#L245-L268). [`system`](https://ruby-doc.org/core-2.6/Kernel.html#method-i-system) already supports passing an env to it, and backticks would need to be replaced with [`Open3.capture2e`](https://ruby-doc.org/stdlib-2.5.3/libdoc/open3/rdoc/Open3.html#method-c-capture2e) or something similar.

Reply:
I suspect that `rails` might need to learn to recognise such env-like parameters, at least while we're transitioning subcommands from rake to thor, even if we mark them as deprecated along the way. (I'm a little surprised we haven't hit this yet, but imagine it'll be more noticeable as we cover more commands, that are run in more varied circumstances.)

As implied by the mention of deprecation, though, I'd prefer we didn't use that as our internal mechanism. 

While I always prefer a many-args strategy for running subcommands, including passing env "properly", I'm not sure it'd work reliably here: I don't recall _when_ it comes into play, but that `#{sudo}` looks ominous. 

All of that said, though, if we know we're running a `rails` subcommand, might it be worth considering invoking `Rails::Command`-type stuff directly?

Reply:
Thanks, I think your concern is correct. 

Originally, `generate` action did not recognize `env` (document has no explanation). And since I can not think of a use case that needs to specify `env` when executing generator, I think there is no need to support that.
Since processing to eliminate `RAILS_ENV` is necessary, I will merge this PR first, especially if there is no objection. 

I would like to consider separately about the behavior when `rails` action is executed.